hack/references/generate.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"cuelang.org/go/cue"
+	"github.com/oam-dev/kubevela/apis/types"
+	mycue "github.com/oam-dev/kubevela/pkg/cue"
+	"github.com/oam-dev/kubevela/pkg/plugins"
+)
+
+const BaseRefPath = "docs/en/developers/references"
+
+type ReferenceMarkdown struct {
+	CapabilityName string `json:"capabilityName"`
+	CapabilityType string `json:"capabilityType"`
+	File           *os.File
+}
+type Parameter struct {
+	types.Parameter `json:",inline,omitempty"`
+	// PrintableType is same to `parameter.Type` which could be printable
+	PrintableType string `json:"printableType"`
+	// Depth marks the depth for calling of function `parseParameters`
+	Depth *int `json:"depth"`
+}
+
+var refContent string
+var recurseDepth *int
+
+func main() {
+	var capabilityType string
+	var specificationType string
+	caps, err := plugins.LoadAllInstalledCapability()
+	if err != nil {
+		fmt.Printf("failed to generate reference docs for all capabilites: %s", err)
+		os.Exit(-1)
+	}
+	for _, c := range caps {
+		if c.Type == "workload" {
+			capabilityType = "workload-types"
+			specificationType = "workload type"
+
+		} else if c.Type == "trait" {
+			capabilityType = "traits"
+			specificationType = "trait"
+		}
+
+		fileName := fmt.Sprintf("%s.md", c.Name)
+		filePath := filepath.Join(BaseRefPath, capabilityType, fileName)
+		f, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE, 0600)
```

Comment:
```suggestion
		f, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE, 0644)
```

这里输出的应该是文档吧？应该用0644 让更广范围的人看到. 忽略安全检查那个golint就好了。